simulation/agents.py

```diff
@@ -619,14 +619,14 @@ def proba_of_becoming_heat_pump_aware_required_to_reach_campaign_target(
         self, model
     ) -> float:
         return (
-            model.campaign_target_heat_pump_awareness - model.heat_pump_awareness
-        ) / (1 - model.heat_pump_awareness)
+            model.campaign_target_heat_pump_awareness
+            - model.heat_pump_awareness_at_timestep
+        ) / (1 - model.heat_pump_awareness_at_timestep)
 
     def update_heat_pump_awareness(self, model) -> None:
-
         if (
             InterventionType.HEAT_PUMP_CAMPAIGN in model.interventions
-            and model.current_datetime == model.heat_pump_awareness_campaign_date
+            and model.current_datetime >= model.heat_pump_awareness_campaign_date
```

Comment:
How confident are you with this implementation? Is it worth having a test to test this logic? Up to you to decide!